modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1261,6 +1263,36 @@ public int pageSize() {
         return dsCfg.getPageSize();
     }
 
+    /** Dumps related partition files and index file to diagnostic dir. */
+    public void dumpPartitionFiles(int grpId, long... pageIds) {
+        try {
+            String path = "db/dump/" + igniteCfg.getConsistentId() + "/" + grpId;
+            File dumpDir = U.resolveWorkDirectory(igniteCfg.getWorkDirectory(), path, false);
+
+            CacheStoreHolder pageStores = idxCacheStores.get(grpId);
+
+            FilePageStore idxStore = (FilePageStore) pageStores.idxStore;
+
+            U.copy(new File(idxStore.getFileAbsolutePath()), new File(dumpDir, INDEX_FILE_NAME), false);
+
+            Set<Integer> parts = new HashSet<>();
+            for (long pageId : pageIds) {
+                parts.add(PageIdUtils.partId(pageId));
+            }
+
+            for (Integer part : parts) {
+                if (part >= pageStores.partStores.length)//skip index parts
```

Comment:
I don't understand this condition and this comment, can you elaborate?

Reply:
Sure, if error occurred in reading index file, then part = 65535, which outbound of data partitions values.